### PR TITLE
Convert binlog.py to go version

### PR DESCRIPTION
**Pull request by arindamnayak**
Ported [binlog.py](https://github.com/planetscale/vitess/blob/master/test/binlog.py) to go version.

### CODE REVIEW

**Review by deepthi — COMMENTED** [review]
Can you write a description for each test case similar to how you already have for `TestCharset`? Maybe also rename `TestMain` to `TestBinlog`? or whatever is appropriate.

Also requires a rebase/merge to get unit tests passing.

File: go/test/endtoend/binlog/binlog_test.go [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+		os.Exit(exitcode)
+	}
+}
+
+//  Insert something that will replicate incorrectly if the charset is not
+//  propagated through binlog streamer to the destination.
+
+//  Vitess tablets default to using utf8, so we insert something crazy and
+//  pretend it's latin1. If the binlog player doesn't also pretend it's
+//  latin1, it will be inserted as utf8, which will change its value.
```
Comment (deepthi) [inline go/test/endtoend/binlog/binlog_test.go:259]: +1. Nice documentation.

**Comment by arindamnayak** [comment]
> Can you write a description for each test case similar to how you already have for `TestCharset`? Maybe also rename `TestMain` to `TestBinlog`? or whatever is appropriate.
> 
> Also requires a rebase/merge to get unit tests passing.

Update the documentations. `TestMain` contains all setup related code, we can't customize that name as it accepts `m *testing.M` . 

It depends on a upstream PR (initial sharding PR), once that is merged and if this PR looks good to you, I will create PR against upstream, where all these unit test case failure will be fixed automatically.

**Review by deepthi — APPROVED** [review]
LGTM

**Comment by ajeetj** [comment]
LGTM, merging in tal_test_master

**Pull request merged**